deep_qa/data/instances/sequence_tagging/verb_semantics_instance.py

```diff
@@ -0,0 +1,173 @@
+
+        tag_one_hot_list = []
+        tag_indices = [data_indexer.get_word_index(tag, namespace='tags') for tag in self.label[1]]
+        for tag_index in tag_indices:
+            # We subtract 2 here to account for the unknown and padding tokens that the DataIndexer
+            # uses.
+            tag_one_hot = numpy.zeros(data_indexer.get_vocab_size(namespace='tags') - 2)
+            tag_one_hot[tag_index - 2] = 1
+            tag_one_hot_list.append(tag_one_hot)
+
+        return (state_one_hot, tag_one_hot_list)
+
+    @classmethod
+    @overrides
+    def read_from_line(cls, line: str):
+        """
+        Reads a VerbSemanticsInstance object from a line.  The format is as follows:
+        [sentence][tab][verb][tab][entity][tab][stateChangeLabel][tab][arg1][tab][arg2]
+
+        sentence: this is pre-tokenized with tokens seperated by "####"
+        verb: startIndex,endIndex of verb in the sentence
+        entity: startIndex,endIndex of entity in the sentence
+        state_change_label: one label from {CREATE, DESTROY, MOVE}
+        arg1: startIndex,endIndex of arg1 in the sentence
+        arg2: startIndex,endIndex of arg2 in the sentence
+        """
+        fields = line.split("\t")
+
+        if len(fields) == 6:
+            sentenceStr, verb_str, entity_str, state_change_label, arg1_str, arg2_str = fields
+            sentence = sentenceStr.split("####")
+            v_parts = verb_str.split(",")
+            verb = (int(v_parts[0]), int(v_parts[1]))
+            e_parts = entity_str.split(",")
+            entity = (int(e_parts[0]), int(e_parts[1]))
+            a1_parts = arg1_str.split(",")
+            arg1 = (int(a1_parts[0]), int(a1_parts[1]))
+            a2_parts = arg2_str.split(",")
+            arg2 = (int(a2_parts[0]), int(a2_parts[1]))
+            return cls(sentence, verb, entity, state_change_label, arg1, arg2)
+        else:
+            raise RuntimeError("Unrecognized line format: " + line)
+
+
+class IndexedVerbSemanticsInstance(IndexedInstance):
+
+    def __init__(self,
+                 sentence: List[int],
+                 verb: List[int],
+                 entity: List[int],
+                 label: Tuple[int, List[int]],
+                 index: int=None):
+
+        super(IndexedVerbSemanticsInstance, self).__init__(label, index)
+        self.sentence = sentence
+        self.verb = verb
+        self.entity = entity
+
+    @classmethod
+    @overrides
+    def empty_instance(cls):
+        return IndexedVerbSemanticsInstance([], [], [], label=None, index=None)
+
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:
+        """
+        This simple IndexedInstance only has one padding dimension: word_indices.
+        """
+        return self._get_word_sequence_lengths(self.sentence)
+
+    @overrides
+    def pad(self, padding_lengths: Dict[str, int]):
+        self.sentence = self.pad_word_sequence(self.sentence, padding_lengths)
```

Comment:
If you represent the verb and entity the way that I suggested above, you'll also need to pad them correctly here, and you need to pad the label.  You'd do all of those with something like [this](https://github.com/allenai/deep_qa/blob/6a0039822f9f2c727151b6143e2214d331bdc111/deep_qa/data/instances/sequence_tagging/tagging_instance.py#L74-L77).